internal/consensus/replay_test.go

```diff
@@ -594,6 +595,8 @@ func TestHandshakeReplayAll(t *testing.T) {
 
 	sim := setupSimulator(ctx, t)
 
+	t.Cleanup(leaktest.Check(t))
```

Comment:
Wouldn't you want these to happen _after_ the simulator cleanup?

Reply:
So the context is cancelled (in these functions defers) and then cleanups run LIFO, with this one being the first one to run, and effectively ensures that there are no threads running before the rest of the cleanup happens. I'm *hoping* that this means that by the time the cleanup function that removes the tempdir, there can't be a thread still alive that could mess with any of the files. 

Doing it in the other order would mean that the cleanup of the tempdir would happen before the leakcheck, which means we would get the existing errors rather than leak check failures.

Reply:
Ah, okay, I guess that makes sense. 🤞🏼